pycqed/analysis_v2/analysis_daemon.py

```diff
@@ -0,0 +1,107 @@
+
+
+import time
+import traceback
+import logging
+import os
+import matplotlib.pyplot as plt
+log = logging.getLogger(__name__)
+log.setLevel(logging.INFO)
+from pycqed.analysis import analysis_toolbox as a_tools
+
+class AnalysisDaemon:
+    def __init__(self, t_start=None, start=True):
+        self.t_start = t_start
+        self.last_ts = None
+        self.poll_interval = 10  # seconds
+        self.errs = []
+        self.job_errs = []
+        if start:
+            self.start()
+
+    def start(self):
+        self.last_ts = a_tools.latest_data(older_than=self.t_start,
+                                           return_path=False,
+                                           return_timestamp=True, n_matches=1)[0]
+        self.run()
+
+    def run(self):
+        try:
+            while (True):
+                self.check_job()
+                for i in range(self.poll_interval):
+                    time.sleep(1)
```

Comment:
why not just `time.sleep(self.poll_interval)`?

Reply:
because you can KeyboardInterrupt only at the end of a sleep

Reply:
Yes as Christoph points out, time.sleep() does not release the lock and therefore exceptions cannot be raised during the sleeping